src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -210,26 +167,37 @@ public HttpResponseMessage RotateDocumentPages(PostedDataEntity postedData)
                 var documentGuid = postedData.guid;
                 var pageNumber = postedData.pages[0];
 
-                string cachePath;
-                ViewerSettings settings = GetViewerSettings(documentGuid, out cachePath);
+                var fileFolderName = Path.GetFileName(documentGuid).Replace(".", "_");
+                string fileCacheSubFolder = Path.Combine(cachePath, fileFolderName);
 
-                File.Delete(Path.Combine(cachePath, $"p{pageNumber}_html.dat"));
-                File.Delete(Path.Combine(cachePath, $"p{pageNumber}_png.dat"));
+                // Delete page cache-files before regenerating with another angle
+                var cacheFiles = Directory.GetFiles(fileCacheSubFolder).Where(f => Path.GetFileName(f).StartsWith($"p{pageNumber}"));
+                cacheFiles.ForEach(f => File.Delete(f));
 
                 PageDescriptionEntity page;
 
-                using (GroupDocs.Viewer.Viewer viewer = new GroupDocs.Viewer.Viewer(documentGuid, GetLoadOptions(postedData.password), settings))
-                {
-                    var currentAngle = GetCurrentAngle(pageNumber, Path.Combine(cachePath, "PagesInfo.xml"));
-                    int newAngle = GetNewAngleValue(currentAngle, postedData.angle);
+                var currentAngle = GetCurrentAngle(pageNumber, Path.Combine(fileCacheSubFolder, "PagesInfo.xml"));
+                int newAngle = GetNewAngleValue(currentAngle, postedData.angle);
+
+                string htmlResourcePrefix = "/viewer/resources/";
+                IViewerCache cache = new FileViewerCache(cachePath, fileCacheSubFolder);
+                HtmlViewer htmlViewer = new HtmlViewer(documentGuid, htmlResourcePrefix, cache, pageNumber, newAngle);
+                PngViewer pngViewer = new PngViewer(documentGuid, htmlResourcePrefix, cache, pageNumber, newAngle);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'pngViewer'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933706)